Update the checkout docs (#189)
mirumee/saleor-docs: docs/developer/checkout.mdx

- `gateway`: the ID of the selected payment gateway (list of the available payment gateways can be fetched from the `Checkout.availablePaymentGateways` field). The selected gateway must support the checkout currency.
- `token`: a client-side generated payment token (if required).
- `amount`: the total amount of this operation.

> - `returnUrl`: URL of a storefront view where user should be redirected after requiring additional actions. Payment with additional actions will not be finished if this field is not provided.

Reviewer: ```suggestion
- `returnUrl`: URL of a storefront view where the user should be redirected after requiring additional actions. Payment with additional actions will not be finished if this field is not provided.
```

## Completing the checkout

The purpose of the operation is to ensure this checkout is correct. To do this, it verifies if:

1. The required addresses are valid.

2. All selected products are in stock (while making the purchase, another user could already buy the last available item).

3. The payment creation succeeded.

> 4. No additional actions are required.

Reviewer 2: What actions? I would add a simple example - "No additional actions are required (e.g. 3DS confirmation)".

Reviewer 3: Some of our community members can ask about it. So adding some examples would be a good idea.

> 4. No additional actions are required by PSP vendor.

Reviewer: ```suggestion
4. No additional actions are required by the PSP.
```

If these are satisfied, the checkout is transformed into an order and the customer receives a confirmation email.

The `checkoutComplete` mutation requires the following input:

- `checkoutId`: the ID of the checkout to complete.

> - `storeSource`: determines whether to store the payment source for future usage.

Reviewer 2: I think it is only supported by Stripe and it would for mentioning here.

Author: AFAIK It is supported by Paymentinterface as we pass it to the payment gateway. In that case, we should include it and explain.

> - `storeSource`: determines whether to store the payment source for future usage.

Reviewer: ```suggestion
- `storeSource`: determines whether to store the payment source for future use.
```

- `redirectUrl`: URL of a view where users should be redirected to see the order details. URL in RFC 1808 format.
- `paymentData`: Client-side generated data required to finalize the payment.

It returns the following output:

- `order`: an `Order` object created from the checkout object.
- `confirmationNeeded`: set to `true` if payment needs to be confirmed before checkout is complete.

> - `confirmationData`: confirmation data used to process additional authorization steps, required by PSP vendor.

Reviewer: ```suggestion
- `confirmationData`: confirmation data used to process additional authorization steps, required by the PSP.
```

  }
}
```

> Here is the example of a `checkoutComplete` mutation where payment gateway requires additional data to finalize the payment:

Reviewer 3: ```suggestion
Here is an example of a `checkoutComplete` mutation where the payment gateway requires additional data to finalize the payment:
```

> Here is an example of a `checkoutComplete` mutation where the payment gateway requires additional data to finalize the payment:

Reviewer 2: ```suggestion
Here is an example of the `checkoutComplete` mutation where the payment gateway requires additional data to finalize the payment using the Adyen payment gateway:
```

```graphql {2-4}
mutation{
  checkoutComplete(
    checkoutId:"Q2hlY2tvdXQ6NWRiZmNmYjctMmFkYi00ZDU4LThmNTMtZjVmYjk0YjJmY2Ew",

>     paymentData:"{\"paymentMetodData\":\"some payment data\"}"){

Reviewer 2: Can we provide a more meaningful example of payment data here? Something that looks like a real-life example? It would be easier to understand what is expected here.

Author: Hhm, not sure fo this. This field contains the data which will be passed to the payment plugin, payment plugin will use it to start processing the payment. It can be some advanced structure or some random string. I can add it but I am not sure if this will be helpful.

>     paymentData:"{\"paymentMethod\": {\"type\": \"scheme\", \"encryptedCardNumber\": \"adyenjs_0_1_18$k7s65M5V0KdPxTErhBIPoMPI8HlC..\", \"encryptedExpiryMonth\": \"adyenjs_0_1_18$p2OZxW2XmwAA8C1Avxm3G9UB6e4..\", \"encryptedExpiryYear\": \"adyenjs_0_1_18$CkCOLYZsdqpxGjrALWHj3QoGHqe+..\", \"encryptedSecurityCode\": \"adyenjs_0_1_18$XUyMJyHebrra/TpSda9fha978+..\", \"holderName\": \"S. Hopper\"}}"){

Reviewer 2: This should be enough to give the idea what kind of data may be needed here: 

```suggestion
    paymentData:"{\"paymentMethod\": {\"type\": \"scheme\", \"holderName\": \"S. Hopper\", ... }}"){
```

     confirmationNeeded
     confirmationData
     order
     checkoutErrors{
       field
       message
       code
     }
}
```

A response would look like this:
```json
{
  "data": {
    "checkoutComplete": {
      "order": {
        "id": "T3JkZXI6MjU=",
        "status": "UNFULFILLED"
      },
      "confirmationNeeded": false,
      "confirmationData": null,
      "checkoutErrors": []
    }
  }
}
```

> The example of a complete mutation where payment gateway requires additional steps from the user:

Reviewer 3: ```suggestion
An example of a complete mutation where the payment gateway requires additional steps from the user:
```
